src/group/groupService.ts

```diff
@@ -244,3 +224,159 @@ const addGroup = (group: Group) => {
 
 export const saveGroup = ({ group }: { group: Group }) =>
   group.id ? updateGroup(group) : addGroup(group);
+
+export const leaveGroup = (
+  input: GroupMembershipDeleteMutationInput,
+  user: User | null
+) => {
+  const query = graphql(`
+    mutation deleteGroupMembership(
+      $input: GroupMembershipDeleteMutationInput!
+      $accountEmail: String!
+    ) {
+      deleteGroupMembership(input: $input) {
+        group {
+          ...groupFields
+          ...groupDataEntries
+          ...groupMembershipListWithMembersSample
+        }
+        errors
+      }
+    }
+  `);
+  return terrasoApi
+    .requestGraphQL(query, {
+      input,
+      accountEmail: user!.email,
+    })
+    .then(resp => resp.deleteGroupMembership.group)
+    .then(extractGroup);
+};
+
+export const joinGroup = (
+  { groupSlug }: { groupSlug: string },
+  user: User | null
+) => {
+  const query = graphql(`
+    mutation joinGroup(
+      $input: GroupMembershipSaveMutationInput!
+      $accountEmail: String!
+    ) {
+      saveGroupMembership(input: $input) {
+        group {
+          ...groupFields
+          ...groupDataEntries
+          ...groupMembershipListWithMembersSample
+        }
+        errors
+      }
+    }
+  `);
+  return terrasoApi
+    .requestGraphQL(query, {
+      input: {
+        groupSlug,
+        userEmails: [user!.email],
+        userRole: ROLE_MEMBER,
+      },
+      accountEmail: user!.email,
+    })
+    .then(resp => resp.saveGroupMembership.group)
+    .then(extractGroup);
+};
+
+export const leaveGroupFromList = (
+  input: GroupMembershipDeleteMutationInput
+) => {
+  const query = graphql(`
+    mutation leaveGroupFromList($input: GroupMembershipDeleteMutationInput!) {
+      deleteGroupMembership(input: $input) {
+        group {
+          ...groupFields
+          ...groupMembershipList
+        }
+        errors
+      }
+    }
+  `);
+  return terrasoApi
+    .requestGraphQL(query, {
+      input,
+    })
+    .then(resp => resp.deleteGroupMembership.group)
+    .then(extractGroup);
+};
+
+export const joinGroupFromList = (
```

Comment:
question: why is this `fromList`?

Reply:
@paulschreiber We need different data to be fetched from the api after the join request, so this is the request needed when the user is joining from the group list page

Reply:
@paulschreiber The name is clear to me, let me know what is your suggestion for a better name

Reply:
`joinGroupFromListPage` perhaps?